### PR TITLE
Fix implementation imports outside of lib

### PR DESCRIPTION
**Pull request by goderbauer**
Work towards https://github.com/dart-lang/linter/issues/4859

There are libraries outside a `lib/` directory, which violate `implementation_imports`.

### CODE REVIEW

File: lib/web_ui/dev/test_platform.dart [hunk 1]
```diff
@@ -22,16 +22,13 @@ import 'package:shelf_web_socket/shelf_web_socket.dart';
...
-import 'package:test_core/src/runner/suite.dart';
-import 'package:test_core/src/util/io.dart';
-import 'package:test_core/src/util/stack_trace_mapper.dart';
+import 'package:test_core/backend.dart' hide Compiler;
+import 'package:test_core/src/runner/environment.dart'; // ignore: implementation_imports
+import 'package:test_core/src/runner/platform.dart'; // ignore: implementation_imports
+import 'package:test_core/src/runner/plugin/platform_helpers.dart'; // ignore: implementation_imports
+import 'package:test_core/src/runner/runner_suite.dart'; // ignore: implementation_imports
+import 'package:test_core/src/util/io.dart'; // ignore: implementation_imports
+import 'package:test_core/src/util/stack_trace_mapper.dart'; // ignore: implementation_imports
```
Comment (ditman) [inline lib/web_ui/dev/test_platform.dart:31]: All these `ignore`s are suspicious... this code might need some extra cleanup.

**Review by ditman — APPROVED** [review]
Thanks for tackling this @goderbauer!

File: lib/web_ui/dev/test_platform.dart (hunk 1 above)
Comment (goderbauer) [reply lib/web_ui/dev/test_platform.dart:31]: Yeah, the API surface of test_core is .. weird.

Comment (goderbauer) [reply lib/web_ui/dev/test_platform.dart:31]: I am happy to link the issue from the code in a TODO here.

Comment (ditman) [reply lib/web_ui/dev/test_platform.dart:31]: https://github.com/flutter/flutter/issues/143599

**Review by srawlins — APPROVED** [review]
Thanks a million!

**Pull request merged**